### PR TITLE
feat: TexturePacker atlas can be generated from device's file

### PR DESCRIPTION
**Pull request by gnarhard**
# Description

TexturePacker atlas can be generated from device's file.

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [ ] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

File: packages/flame_texturepacker/lib/atlas/texture_packer_atlas.dart [hunk 1]
```diff
@@ -19,8 +21,20 @@ class TexturePackerAtlas {
...
   /// returns a new [TexturePackerAtlas].
-  static Future<TexturePackerAtlas> load(String path) async {
-    final atlasData = await _load(path);
+  /// If [fromStorage] is true, the atlas will be loaded from the device's
+  /// storage instead of the assets folder.
+  static Future<TexturePackerAtlas> load(
+    String path, {
+    bool fromStorage = false,
+  }) async {
+    late final _TextureAtlasData atlasData;
```
Comment (spydon) [inline packages/flame_texturepacker/lib/atlas/texture_packer_atlas.dart:30]: This doesn't have to be late, the compiler figures out that it is always set in the if-else

File: packages/flame_texturepacker/lib/atlas/texture_packer_atlas.dart [hunk 2]
```diff
@@ -51,27 +65,64 @@ class TexturePackerAtlas {
...
+/// Otherwise, it will be loaded from the assets folder.
+/// Returns a [_TextureAtlasData] containing the pages and regions.
+Future<_TextureAtlasData> _parse(
+  String path, {
+  required bool fromStorage,
+}) async {
   final pages = <Page>[];
   final regions = <Region>[];
-  final fileAsString = await Flame.assets.readFile(path);
+  late final String fileAsString;
```
Comment (spydon) [inline packages/flame_texturepacker/lib/atlas/texture_packer_atlas.dart:99]: Same here

**Review by spydon — COMMENTED** [review]
Overall lgtm!
Rotations were not supported in here, right?

**Comment by gnarhard** [comment]
The data to rotate the image or not is still getting set when we parse through the atlas and build a list of TexturePackerSprites. Within the TexturePackerSprite class, I ripped out the render method and anything related to actually performing a rotation.

**Review by spydon — APPROVED** [review]
Lgtm

**Pull request merged**